### PR TITLE
Experiments

### PR DESCRIPTION
**Pull request by Michalski-Artur**
- [x] Extend experiments by aggregating wins, loses and draws
- [x] Improve mcts by preventing expanding moves which lead to opponent win

### CODE REVIEW

File: experiments.py [hunk 1]
```diff
@@ -80,11 +83,15 @@ def print_scores(self):
...
+            print(f"Scores: {player1['player_type']}  {player1['score']}:{player2['score']}  {player2['player_type']}")
+            print(f"Wins: {player1['player_type']}  {player1['wins']}:{player2['wins']}  {player2['player_type']}")
+            print(f"Draws: {player1['player_type']}  {player1['draws']}:{player2['draws']}  {player2['player_type']}")
+            print(f"Loses: {player1['player_type']}  {player1['loses']}:{player2['loses']}  {player2['player_type']}")
+            print()
 
 
 def main():
-    experiments = Experiments(50)
+    experiments = Experiments(5)
```
Comment (holdenkold) [inline experiments.py]: please change back to 50

File: experiments.py [hunk 2]
```diff
@@ -80,11 +83,15 @@ def print_scores(self):
         for game in self.global_score:
             player1 = game['first_player']
             player2 = game['second_player']
-            print(f"{player1['player_type']}  {player1['score']}:{player2['score']}  {player2['player_type']}")
+            print(f"Scores: {player1['player_type']}  {player1['score']}:{player2['score']}  {player2['player_type']}")
+            print(f"Wins: {player1['player_type']}  {player1['wins']}:{player2['wins']}  {player2['player_type']}")
+            print(f"Draws: {player1['player_type']}  {player1['draws']}:{player2['draws']}  {player2['player_type']}")
+            print(f"Loses: {player1['player_type']}  {player1['loses']}:{player2['loses']}  {player2['player_type']}")
```
Comment (holdenkold) [inline experiments.py:93]: please add two empty lines to separate results from other games

File: experiments.py [hunk 3]
```diff
@@ -63,10 +69,7 @@ def run_experiments(self):
         minimax_config = MinimaxConfiguration(6)
         self.players = [
             MctsPlayer(mcts_config), 
-            Ucb1TunedPlayer(mcts_config),
```
Comment (holdenkold) [inline experiments.py:66]: probably it wasn't intentional delete

**Review by holdenkold — APPROVED** [review]

File: experiments.py (hunk 3 above)
Comment (Michalski-Artur) [reply experiments.py:66]: Nope, forgot to rollback change after experiments. Done!

File: experiments.py (hunk 2 above)
Comment (Michalski-Artur) [reply experiments.py:93]: Sure, done!

File: experiments.py (hunk 1 above)
Comment (Michalski-Artur) [reply experiments.py]: Done!

**Pull request merged**